### PR TITLE
Refine exception code

### PR DESCRIPTION
**Pull request by qiaojialin**

### CODE REVIEW

File: example/rocketmq/src/main/java/org/apache/iotdb/rocketmq/RocketMQConsumer.java [hunk 1]
```diff
@@ -120,7 +120,7 @@ public void prepareConsume() throws MQClientException {
             new String(msg.getBody())));
         try {
           insert(new String(msg.getBody()));
-        } catch (IoTDBConnectionException e) {
+        } catch (Exception e) {
           logger.error(e.getMessage());
```
Comment (jixuan1989) [inline example/rocketmq/src/main/java/org/apache/iotdb/rocketmq/RocketMQConsumer.java]: it is better to throw MQClientException

File: jdbc/src/main/java/org/apache/iotdb/jdbc/IoTDBStatement.java [hunk 2]
```diff
@@ -258,41 +257,27 @@ private boolean executeSQL(String sql) throws TException, SQLException {
...
-        List<Integer> result = execResp.getResult();
-        int len = result.size();
-        int[] updateArray = new int[len];
-        for (int i = 0; i < len; i++) {
-          updateArray[i] = result.get(i);
-        }
-        exception = new BatchUpdateException(execResp.getStatus().getStatusType().getMessage(),
-            updateArray);
+    if (execResp.statusList == null) {
+
```
Comment (jixuan1989) [inline jdbc/src/main/java/org/apache/iotdb/jdbc/IoTDBStatement.java]: ??

File: jdbc/src/main/java/org/apache/iotdb/jdbc/IoTDBStatement.java [hunk 3]
```diff
@@ -258,41 +257,27 @@ private boolean executeSQL(String sql) throws TException, SQLException {
...
+    }
+    int[] result = new int[execResp.statusList.size()];
+    boolean allSuccess = true;
+    String message = "";
+    for (int i = 0; i < result.length; i++) {
+      result[i] = execResp.statusList.get(i).code;
+      if (result[i] != TSStatusCode.SUCCESS_STATUS.getStatusCode()) {
+        allSuccess = false;
+        message = execResp.statusList.get(i).message;
```
Comment (jixuan1989) [inline jdbc/src/main/java/org/apache/iotdb/jdbc/IoTDBStatement.java]: so, only keep the final message?

File: server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java [hunk 4]
```diff
@@ -537,7 +540,7 @@ public void insert(InsertPlan insertPlan) throws QueryProcessException {
       while (loc < batchInsertPlan.getRowCount()) {
         long time = batchInsertPlan.getTimes()[loc];
         long curTimePartition = StorageEngine.fromTimeToTimePartition(time);
-        results[loc] = TSStatusCode.SUCCESS_STATUS.getStatusCode();
+        results[loc] = RpcUtils.getStatus(TSStatusCode.SUCCESS_STATUS);
```
Comment (jixuan1989) [inline server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java]: overhead if all are successful. (too many TSStatus instances.. right?)

File: server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java [hunk 5]
```diff
@@ -886,11 +837,8 @@ public TSFetchResultsResp fetchResults(TSFetchResultsReq req) {
           QueryResourceManager.getInstance().endQuery(req.queryId);
           queryId2DataSet.remove(req.queryId);
         }
-        TSFetchResultsResp resp =
-            getTSFetchResultsResp(
-                getStatus(
-                    TSStatusCode.SUCCESS_STATUS,
-                    "FetchResult successfully. Has more result: " + hasResultSet));
+        TSFetchResultsResp resp = RpcUtils.getTSFetchResultsResp(
+            TSStatusCode.SUCCESS_STATUS, "FetchResult successfully. Has more result: " + hasResultSet);
```
Comment (jixuan1989) [inline server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java]: do we really need such a long string as the message?

File: server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java [hunk 6]
```diff
@@ -910,18 +858,16 @@ public TSFetchResultsResp fetchResults(TSFetchResultsReq req) {
           queryId2DataSet.remove(req.queryId);
         }
         TSFetchResultsResp resp =
-            getTSFetchResultsResp(
-                getStatus(
-                    TSStatusCode.SUCCESS_STATUS,
-                    "FetchResult successfully. Has more result: " + hasResultSet));
+            RpcUtils.getTSFetchResultsResp(TSStatusCode.SUCCESS_STATUS,
+                "FetchResult successfully. Has more result: " + hasResultSet);
```
Comment (jixuan1989) [inline server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java]: The same with Line 841.

File: service-rpc/src/main/thrift/rpc.thrift [hunk 7]
```diff
@@ -19,18 +19,11 @@
 namespace java org.apache.iotdb.service.rpc.thrift
 
 // The return status code and message in each response.
-struct TSStatusType {
+struct TSStatus {
   1: required i32 code
   2: required string message
```
Comment (jixuan1989) [inline service-rpc/src/main/thrift/rpc.thrift]: maybe the message can be optional.

Comment (qiaojialin) [reply service-rpc/src/main/thrift/rpc.thrift]: fixed

File: server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java (hunk 5 above)
Comment (qiaojialin) [reply server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java]: This message is not needed, removed

File: server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java (hunk 6 above)
Comment (qiaojialin) [reply server/src/main/java/org/apache/iotdb/db/service/TSServiceImpl.java]: removed message

File: server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java (hunk 4 above)
Comment (qiaojialin) [reply server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java]: Yes, I create a public static TSStatus SUCCESS_STATUS = new TSStatus(TSStatusCode.SUCCESS_STATUS.getStatusCode());
in RpcUtils and reuse this.

File: jdbc/src/main/java/org/apache/iotdb/jdbc/IoTDBStatement.java (hunk 3 above)
Comment (qiaojialin) [reply jdbc/src/main/java/org/apache/iotdb/jdbc/IoTDBStatement.java]: Yes, the more detailed message could be found in the result.

File: jdbc/src/main/java/org/apache/iotdb/jdbc/IoTDBStatement.java (hunk 2 above)
Comment (qiaojialin) [reply jdbc/src/main/java/org/apache/iotdb/jdbc/IoTDBStatement.java]: removed

File: server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java (hunk 4 above)
Comment (qiaojialin) [reply server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java]: If all are successful, I will return one TSStatus in TSServiceImpl line 1194

if (isAllSuccessful) {
        if (logger.isDebugEnabled()) {
          logger.debug("Insert one RowBatch successfully");
        }
        return RpcUtils.getTSBatchExecuteStatementResp(TSStatusCode.SUCCESS_STATUS);
      } else {

File: example/rocketmq/src/main/java/org/apache/iotdb/rocketmq/RocketMQConsumer.java (hunk 1 above)
Comment (qiaojialin) [reply example/rocketmq/src/main/java/org/apache/iotdb/rocketmq/RocketMQConsumer.java]: The  MessageListenerOrderly interface doesn't allow to throw an exception

**Pull request merged**